lib/insights/api/common/rbac/access.rb

```diff
@@ -39,26 +52,33 @@ def self.enabled?
 
           private
 
-          def ids
-            @ids ||= @acl.each_with_object([]) do |item, ids|
-              item.resource_definitions.each do |rd|
-                next unless rd.attribute_filter.key == 'id'
-                next unless rd.attribute_filter.operation == 'equal'
+          def scope?(app_name, resource, verb, scope)
+            regexp = create_regexp(app_name, resource, verb)
+            @acls.any? do |item|
+              regexp.match?(item.permission) && scope_matches?(item, scope)
+            end
+          end
 
-                ids << rd.attribute_filter.value
-              end
+          def scope_matches?(item, scope)
+            item.resource_definitions.any? do |rd|
+              rd.attribute_filter.key == 'scope' &&
+                rd.attribute_filter.operation == 'equal' &&
+                rd.attribute_filter.value == scope
             end
           end
 
-          def owner_scope_filter?
-            @acl.any? do |item|
-              item.resource_definitions.any? do |rd|
-                rd.attribute_filter.key == 'owner' &&
-                  rd.attribute_filter.operation == 'equal' &&
-                  rd.attribute_filter.value == '{{username}}'
+          def all_scopes(item)
```

Comment:
since each scope can be one of three possible values, and admin > group > user, can a permission in reality have more than one scope defined in the definition?

Reply:
@bzwei Yes, the attribute filter is an array. Plus the permission is coming from different roles.

Approval Administrator
                   -> approval.workflows.read
                          sope = admin
Approval Approver
                  -> approval:workflows:read
                         scope = group

Once you make a get_principal_access call it dissolves the roles and you are left with
-> approval:workflows:read
                          sope = admin
-> approval:workflows:read
                         scope = group

When the API calls in the above scenario with
scopes('workflows',read') the result will be ['admin','group']